server/controllers/finance/accounts/references.compute.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * Accounts References Computations
+ */
+const Q = require('q');
+const db = require('../../../lib/db');
+
+/**
+ * @method computeAllAccountReference
+ *
+ * @description
+ * compute value of all account references and returns an array of all accounts reference
+ * with their debit, credit and balance
+ *
+ * @param {number} periodId - the period needed
+ */
+function computeAllAccountReference(periodId) {
+  const glb = {};
+
+  // get fiscal year information for the given period
+  const queryFiscalYear = `
+    SELECT fy.id, p.number AS period_number FROM fiscal_year fy
+    JOIN period p ON p.fiscal_year_id = fy.id
+    WHERE p.id = ?
+  `;
+
+  // get all references
+  const queryAccountReferences = `
+    SELECT id, abbr, description, is_amo_dep FROM account_reference;
+  `;
+
+  return db.one(queryFiscalYear, [periodId])
+    .then(fiscalYear => {
+      glb.fiscalYear = fiscalYear;
+
+      return db.exec(queryAccountReferences);
```

Comment:
It doesn't look like this query depends on the previous query.  You could speed it up by combining both queries and wrapping them in `Q.all()`.